library/x509write_crt.c

```diff
@@ -178,10 +188,24 @@ int mbedtls_x509write_crt_set_subject_key_identifier( mbedtls_x509write_cert *ct
     MBEDTLS_ASN1_CHK_ADD( len,
                 mbedtls_pk_write_pubkey( &c, buf, ctx->subject_key ) );
 
```

Comment:
I realize this is out of the intended scope of the current task, but `mbedtls_x509write_crt_set_authority_key_identifier` and `mbedtls_x509write_crt_set_subject_key_identifier` are very similar and we're making an identical change to the two of them that nontrivially increases their size. So please create an auxiliary function for the common part (write pub key and its SHA-1).